src/RoomDirectoryModel.h

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <QAbstractListModel>
+#include "ui/FlatButton.h"
+
+#include <mtxclient/http/errors.hpp>
+#include "MatrixClient.h"
+
+#include <vector>
+
+class RoomDirectoryModel : public QAbstractListModel
+{
+public: 
+    explicit RoomDirectoryModel
+    (const std::string &s = "", QObject *parent = nullptr);
+
+    enum Roles {
+        // ??? TODO ???
+    };
+    QHash<int, QByteArray> roleNames() const override;
+
+    QVariant data(const QModelIndex &index, int role) const override;
+
+    inline int rowCount(const QModelIndex &parent = QModelIndex()) const override
+    {
+            (void)parent; // ???copied from UserModel - (why) do we need this???
```

Comment:
Because we don't use the parent, but need to provide an overload that takes a defaulted parent :3 The void casts is a way to say this is unused, another one is the [[maybe_unused]] attribute.